### PR TITLE
[Exam] Prevent NPE in quiz exam summary and complaint interactions component

### PR DESCRIPTION
**Pull request by sascha11110**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I documented the TypeScript code using JSDoc style.

### Motivation and Context
(1)
image

(2)
image

### Description
- [x] I've ensured that there is no chance of a null pointer / index out of bound exception in the `quiz-exam-summary.component`
- [x] I've ensured that there won't be a null pointer exception if `this.result` is not set in `isTimeOfComplaintValid()` in the `complaint-interactions.component`

### CODE REVIEW

File: src/main/webapp/app/exam/participate/summary/exercises/quiz-exam-summary/quiz-exam-summary.component.ts [hunk 1]
```diff
@@ -11,6 +10,7 @@ import { ShortAnswerSubmittedAnswer } from 'app/entities/quiz/short-answer-submi
 import { QuizExercise } from 'app/entities/quiz/quiz-exercise.model';
 import { QuizExerciseService } from 'app/exercises/quiz/manage/quiz-exercise.service';
 import { Exam } from 'app/entities/exam.model';
+import { ArtemisServerDateService } from 'app/shared/server-date.service';
```
Comment (TobiasPr) [inline src/main/webapp/app/exam/participate/summary/exercises/quiz-exam-summary/quiz-exam-summary.component.ts:13]: why did you include the service?

File: src/main/webapp/app/exam/participate/summary/exercises/quiz-exam-summary/quiz-exam-summary.component.ts [hunk 2]
```diff
@@ -40,7 +39,7 @@ export class QuizExamSummaryComponent implements OnInit {
     @Input()
     exam: Exam;
 
-    constructor(private exerciseService: QuizExerciseService) {}
+    constructor(private exerciseService: QuizExerciseService, private serverDateService: ArtemisServerDateService) {}
```
Comment (TobiasPr) [inline src/main/webapp/app/exam/participate/summary/exercises/quiz-exam-summary/quiz-exam-summary.component.ts:42]: why did you include the service?

**Comment by TobiasPr** [comment]
> I checked on ts3. I did an exam with a MC, DnD and SA quiz and did not evaluate them until the review period began - everything worked fine. Then I evaluated them and it was also correctly displayed. Good job

Did not see your latest changes, will do a review again once builded

**Comment by sascha11110** [comment]
@TobiasPr Thanks for reviewing! Actually, I couldn't reproduce the errors locally and on the test server. They just appeared on production. So, hopefully this will fix the problems in production.

**Comment by sascha11110** [comment]
@TobiasPr Could you re-approve? I've just solved a merge conflict.

**Pull request merged**